src/sidebar/components/menu-keyboard-navigation.js

```diff
@@ -8,27 +8,45 @@ function isElementVisible(element) {
   return element.offsetParent !== null;
 }
 
+/**
+ * @template T
+ * @typedef {import("preact/hooks").Ref<T>} Ref
+ */
+
+/**
+ * @typedef MenuKeyboardNavigationProps
+ * @prop {string} [className]
+ * @prop {(e: KeyboardEvent) => any} [closeMenu] - Callback when the menu is closed via keyboard input
+ * @prop {boolean} [visible] - When  true`, sets focus on the first item in the list
+ * @prop {Object} children - Array of nodes which may contain <MenuItems> or any nodes
```

Comment:
scripts/jsdoc-from-proptypes.js has this going to an `Object`, but I do wonder if there is a better type for children.

`{Element[]|Element}` seems a bit closer, but then again, maybe this is all superficial and irrelevant :)

Reply:
Using `Object` for children is fine here I think. You can do something more specific but it would be a bit verbose. Also, it is important not to mix up the DOM Element type with Preact elements (`import(“preact”).JSX.Element`)